filesystem/squashfs/directoryentry.go

```diff
@@ -158,3 +161,39 @@ func (d *directoryEntry) Readlink() (string, error) {
 	}
 	return target, nil
 }
+
+// Open returns an filesystem.File from which you can read the
+// contents of a file.
+//
+// Calling this on anything but a file will return an error.
+//
+// Calling this Open method is more efficient than calling
+// FileSystem.OpenFile as it doesn't have to find the file by
+// traversing the directory entries first.
+func (d *directoryEntry) Open() (filesystem.File, error) {
```

Comment:
I was going to ask, why not have the `mode` arg here... and then realized, "we are talking about a read-only filesystem, so no point to it." Looks good.